### PR TITLE
ROX-20602:  compliance scan config status pk

### PR DESCRIPTION
**Pull request by dashrews78**
## Description

Initially compliance scan configuration status used the cluster id as the PK.  This was a mistake as it prevented tracking the status of multiple scans on the same cluster.  So this PR resolves that issue by creating a deterministic PK based off the cluster and scan configuration.  This ensures that a single cluster can be associated with multiple scan configurations.

## Checklist
- [x] Investigated and inspected CI test results
- [x] Unit test and regression tests added
~- [ ] Evaluated and added CHANGELOG entry if required~
~- [ ] Determined and documented upgrade steps~
~- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs]~(https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### Here I tell how I validated my change

Updated the unit tests to better cover this case.

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

File: central/complianceoperator/v2/scanconfigurations/datastore/datastore.go [hunk 1]
```diff
@@ -61,17 +61,6 @@ func New(scanConfigStore pgStore.Store, scanConfigStatusStore statusStore.Store,
 	return ds
 }
 
-// NewForTestOnly returns an instance of DataStore only for tests.
-func NewForTestOnly(_ *testing.T, scanConfigStore pgStore.Store, scanConfigStatusStore statusStore.Store, clusterDS clusterDatastore.DataStore) DataStore {
```
Comment (dashrews78) [inline central/complianceoperator/v2/scanconfigurations/datastore/datastore.go:65]: Unused.

File: central/complianceoperator/v2/scanconfigurations/datastore/datastore_impl.go [hunk 2]
```diff
@@ -125,7 +126,15 @@ func (ds *datastoreImpl) UpdateClusterStatus(ctx context.Context, scanID string,
 		return errors.Errorf("Unable to find scan configuration id %q", scanID)
 	}
 
+	// Need to build a deterministic ID from clusterID and scanID to ensure we always have the latest status
+	scanUUID, err := uuid.FromString(scanID)
+	if err != nil {
+		return errors.Errorf("Unable to build scan configuration status id based off %q", scanID)
```
Comment (connorgorman) [inline central/complianceoperator/v2/scanconfigurations/datastore/datastore_impl.go:132]: errors.Wrapf so the err is contained

File: central/complianceoperator/v2/scanconfigurations/datastore/datastore_impl.go [hunk 3]
```diff
@@ -125,7 +126,15 @@ func (ds *datastoreImpl) UpdateClusterStatus(ctx context.Context, scanID string,
 		return errors.Errorf("Unable to find scan configuration id %q", scanID)
 	}
 
+	// Need to build a deterministic ID from clusterID and scanID to ensure we always have the latest status
+	scanUUID, err := uuid.FromString(scanID)
```
Comment (connorgorman) [inline central/complianceoperator/v2/scanconfigurations/datastore/datastore_impl.go:130]: I don't know why, but it feels like it should be uuid from cluster and then the scan uuid is being generated from that?

**Review by connorgorman — COMMENTED** [review]
couple comments

File: central/complianceoperator/v2/scanconfigurations/datastore/datastore_impl.go (hunk 3 above)
Comment (dashrews78) [reply central/complianceoperator/v2/scanconfigurations/datastore/datastore_impl.go:130]: I basically flipped a coin on that.  I will flip it to cluster.

**Review by connorgorman — APPROVED** [review]

**Pull request merged**